tt_metal/impl/device/multi_device.cpp

```diff
@@ -52,11 +52,16 @@ DeviceMesh::DeviceMesh(const DeviceGrid& device_grid, const DeviceIds &device_id
         for (int i = 0; i < num_requested_devices; i++) {
             mesh_devices.emplace_back(device_ids[i], managed_devices.at(galaxy_device_ids[i]));
         }
+        this->num_rows = num_rows;
+        this->num_cols = num_cols;
     } else {
         managed_devices = tt::tt_metal::detail::CreateDevices(device_ids, num_command_queues, l1_small_size, trace_region_size);
         for (int i = 0; i < num_requested_devices; i++) {
             mesh_devices.emplace_back(device_ids[i], managed_devices.at(device_ids[i]));
         }
+        // TODO: support concept of rows/cols in other systems
+        this->num_rows = 0;
```

Comment:
why can't we just add this logic unconditionally:

```
        this->num_rows = num_rows;
        this->num_cols = num_cols;
```

so that it works for non-galaxy devices like T3K as well

Reply:
Currently, we don't have the support for rows and cols for T3k, so I set it to 0 to prevent users from calling rows and cols.

Reply:
Why wouldn't this work on T3K? `managed_devices` are not returned in row-major?

Reply:
Yea unfortunately :(

Reply:
Is there an issue for this?

Reply:
https://github.com/tenstorrent/tt-metal/issues/10419

Reply:
Can you make a reference to that here: https://github.com/tenstorrent/tt-metal/pull/10417/files/3dcc92c902afe1e469938ba007c2b50e30895bd9#diff-b1e1f2d15597e5ac54c3b3339d961fa4bd7a3d3aaedf9452329a969b28473c47R100

Otherwise rest of the changes look good to me, thanks!